### DISCUSSION TITLE
Device not recognised

### DISCUSSION THREAD

**Discussion by woepies** [zwave-js/zwave-js#3015] [Q&A] [answered]
I purchased a wall mounted remote a while ago, the ROB_100-004-0 ([link](https://www.robbshop.nl/wandzender-4v-z-wave-robb-smarrt)). After including the device, the ZWave2MQTT control panel reports "Unknown manufacturer 0x0330" and Unknown product 0xa306. The device ID is 816-41734-768 (0x0330-0x0300-0xa306). After some research, I noted that the device seems to be a rebranded [Sunricher](https://products.z-wavealliance.org/products/3783).

The Sunricher configuration file already exists ([here](https://github.com/zwave-js/node-zwave-js/blob/master/packages/config/config/devices/0x0330/zv9001k4-dim-g2.json)). However, the productType in the configuration file is 0x0003, while the zwave alliance webpage states that the product Type ID is 0x0300. Is this a typo or intended?

I am running zwavejs2mqtt in a docker container. I changed the productType from 0x0003 to 0x0300 in the device configuration file (zv9001k4-dim-g2.json) in the docker container. 
`        "devices": [
                {
                        "productType": "0x0300",
                        "productId": "0xa306",
                        "zwaveAllianceId": 3783
                }
`
I excluded the device, restarted the zwavejs2mqtt container and included the device, but same error remains.

What can I do to make ZWaveJS2MQTT to recognise my device?

**Comment by AlCalzone** [comment, marked as answer]
That seems to be a typo - I'll fix it.
For reference, here's how you can test custom files without having to mess around in the docker image: https://zwave-js.github.io/zwavejs2mqtt/#/development/intro?id=testing-custom-devices-config

**Reply by AlCalzone** [reply]
https://github.com/zwave-js/node-zwave-js/pull/3018

**Reply by woepies** [reply]
Thanks for the fix and the link. Greatly appreciated.  I have put the manually updated configuration file in the `<app-root-folder>/store/config` folder. After re-interviewing, the node is not recognised. The logfile mentions "No device config found". I double checked that the id's in the logfile match with the device config file:

2021-07-15T06:02:57.519Z DRIVER « [Node 029] [REQ] [ApplicationCommand]
                                  └─[ManufacturerSpecificCCReport]
                                      manufacturer id: 0x0330
                                      product type:    0x0300
                                      product id:      0xa306

2021-07-15T06:02:57.902Z DRIVER « [Node 029] [REQ] [ApplicationCommand]
                                  └─[VersionCCReport]
                                      library type:      Enhanced Slave
                                      protocol version:  5.3
                                      firmware versions: 1.26
                                      hardware version:  1

No success yet. It feels like I am overlooking something. But what?

**Reply by AlCalzone** [reply]
Can you share the file you placed there? Maybe it has a syntax error, in which case it won't be picked up either.

**Reply by woepies** [reply]
Here is the file. Note that I changed the extension to .txt because the upload of a .json file is not supported. Just to make sure: The file on my system still has the json extension.

[zv9001k4-dim-g2.txt](https://github.com/zwave-js/node-zwave-js/files/6821451/zv9001k4-dim-g2.txt)

**Reply by AlCalzone** [reply]
Hmm that file looks okay. Have you tried restarting zwavejs2mqtt?

**Reply by woepies** [reply]
A restart of zwavejs2mqtt solved the issue! The device is now recognised, and the logfile mentions " [Node 029] User-provided device config loaded". Thanks for your assistance.
